Source/WebSocketImpl_Apple.m

```diff
@@ -0,0 +1,73 @@
+#import <WebSocketImpl_Apple.h>
+
+@interface WebSocket_Impl() <NSURLSessionWebSocketDelegate> {
+    NSURLSessionWebSocketTask *webSocketTask API_AVAILABLE(ios(13.0));
+    NSURLSession *session;
+    // store callbacks here
+    void (^open_callback)();
+    void (^close_callback)();
+    void (^error_callback)();
+    void (^message_callback)();
+}
+@end
+
+@implementation WebSocket_Impl
+
+-(void) open:(NSString *)url on_open:(void (^)(void))on_open on_close:(void (^)(void))on_close on_message:(void (^)(NSString *))on_message on_error:(void (^)(void))on_error {
+    session = [NSURLSession sessionWithConfiguration:[NSURLSessionConfiguration defaultSessionConfiguration] delegate:self delegateQueue:[NSOperationQueue mainQueue]];
+    open_callback = on_open;
+    close_callback = on_close;
+    error_callback = on_error;
+    message_callback = on_message;
+    
+    webSocketTask = [session webSocketTaskWithURL:[NSURL URLWithString:url] ];
+    [webSocketTask resume];
+}
+
+- (void) close {
+    [webSocketTask cancel];
+    [self invalidateAndCancel];
+}
+
+- (void) sendMessage:(NSString *)message {
+    [webSocketTask sendMessage: [[NSURLSessionWebSocketMessage alloc] initWithString:message] completionHandler:^(NSError * _Nullable error) {
+        if (error) {
+            error_callback();
+        } else {
+            NSLog(@"Sent message: %@", message);
```

Comment:
>NSLog(@"Sent message: %@", message); [](http://example.com/codeflow?start=12&length=36)

hmm logging is generally not a bad thing, but this seems to be the only impl that's logging.  Feels to me like we should either log in all 3 impls, or don't log in any of them.

Reply:
Removed logs